local_server/local-server-generate-elevation-profile/src/main/java/com/esri/samples/local_server_generate_elevation_profile/LocalServerGenerateElevationProfileController.java

```diff
@@ -0,0 +1,569 @@
+
+    return maskedRaster;
+  }
+
+  /**
+   * Calculates a camera position and heading angle that is placed perpendicularly to the polyline sketch. 
+   * @return camera with calculated camera position and heading angle
+   */
+  private Camera createCameraFacingElevationProfile() {
+
+    // get the polyline's end point co-ordinates
+    Point endPoint = polyline.getParts().get(0).getEndPoint();
+    var endPointX = endPoint.getX();
+    var endPointY = endPoint.getY();
+    // get the polyline's center point co-ordinates
+    Point centerPoint = polyline.getExtent().getCenter();
+    var centerX = centerPoint.getX();
+    var centerY = centerPoint.getY();
+    // calculate the position of a point perpendicular to the centre of the polyline
+    double lengthX = endPointX - centerX;
+    double lengthY = endPointY - centerY;
+    Point cameraPositionPoint = new Point(centerX + lengthY * 2, centerY - lengthX * 2, 1200,
+      SpatialReferences.getWebMercator());
+
+    // calculate the heading for the camera position so that it points perpendicularly towards the elevation profile
+    double theta;
+    double cameraHeadingPerpToProfile;
+    // account for switching opposite and adjacent depending on angle direction from drawn line
+    if (lengthY < 0) { // accounts for a downwards angle
+      theta = Math.toDegrees(Math.atan((centerX - endPointX) / (centerY - endPointY)));
+      cameraHeadingPerpToProfile = theta + 90;
+    } else { // accounts for an upwards angle
+      theta = Math.toDegrees(Math.atan((centerY - endPointY) / (centerX - endPointX)));
+      // determine if theta is positive or negative, then account accordingly for calculating the angle back from north
+      // and then rotate that value by + or - 90 to get the angle perpendicular to the drawn line
+      double angleFromNorth = (90 - theta);
+      // if theta is positive, rotate angle anticlockwise by 90 degrees, else, clockwise by 90 degrees
+      cameraHeadingPerpToProfile = theta > 0 ? angleFromNorth - 90 : angleFromNorth + 90;
+    }
+    // create a new camera from the calculated camera position point and camera angle perpendicular to profile
+    return new Camera(cameraPositionPoint, cameraHeadingPerpToProfile, 80, 0);
+
+  }
+
+  /**
+   * Displays an information alert and closes the application when dialog is closed.
+   *
+   * @param title header text of the dialog
+   * @param message content text of the dialog
+   */
+  private void showMessage(String title, String message) {
+
+    Platform.runLater(() -> {
+      Alert dialog = new Alert(AlertType.INFORMATION);
+      dialog.initOwner(sceneView.getScene().getWindow());
+      dialog.setHeaderText(title);
+      dialog.setContentText(message);
+      dialog.showAndWait();
+
+      Platform.exit();
+    });
+  }
+
+  /**
+   * Stops and releases all resources used in application.
+   */
+  void terminate() {
+
```

Comment:
Do we need anything to stop local server?  @colinanderson would have a view on best practice here; it might just shutdown on exit, but worth checking

Reply:
have added a stopAsync call in here after discussion with Colin